src/chol.jl

```diff
@@ -25,6 +25,8 @@ if HAVE_CHOLMOD
     chol_upper(cf::CholTypeSparse) = cf.UP
 end
 
+mattype(::Cholesky{T,S}) where {T,S} = S
```

Comment:
Maybe omit it since it's only used once?

Reply:
Sure, but we need a replacement. We need to extract the `S` from `Cholesky{T,S}` but note the `S` may not be the same `S` as for the input to the function that uses `mattype`. (The `convert(Cholesky{T}, c)` might change the `S` type.)

Reply:
It seems on the master branch we use `typeof(c.factors)`.